script/Script.s.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {IBPool} from 'contracts/BPool.sol';
+import {IFaucet} from 'interfaces/IFaucet.sol';
+
+import {Script} from 'forge-std/Script.sol';
+import {Registry} from 'script/Registry.s.sol';
+
+/// @notice This base script is shared across `yarn script:{mainnet|testnet}`
+abstract contract BaseScript is Registry, Script {
+  constructor() Registry(block.chainid) {}
+}
+
+/// @notice This script will be executed by `yarn script:mainnet`
+contract MainnetScript is BaseScript {
+  function run() public {
+    assert(block.chainid == 1);
+    vm.startBroadcast();
+
+    // script logic here
```

Comment:
what do you envision will be done on mainnet?  set up a few pools?

Reply:
up to the dev, i really can't know

Reply:
It would have been useful context for reviewing to know these scripts are specifically meant as templates for future devs to implement new scripts. Was that information somewhere and I missed it? In retrospect, I would've appreciated it, in order or comfyness:

- the git history
- the PRs description
- the description of a linked linear task